pkg/vm/createEvm.go

```diff
@@ -73,54 +54,48 @@ func CreateEvmSubnetConfig(app *application.Avalanche, subnetName string, genesi
 	return genesisBytes, sc, nil
 }
 
-func nextStage(currentState wizardState, direction stateDirection) wizardState {
-	switch direction {
-	case forward:
-		currentState++
-	case backward:
-		currentState--
-	default:
-		return errored
-	}
-	return currentState
-}
-
 func createEvmGenesis(app *application.Avalanche, subnetName string, subnetEVMVersion string) ([]byte, *models.Sidecar, error) {
 	ux.Logger.PrintToUser("creating subnet %s", subnetName)
 
 	genesis := core.Genesis{}
 	conf := params.SubnetEVMDefaultChainConfig
 
-	stage := startStage
+	const (
+		descriptorsState = "descriptors"
+		feeState         = "fee"
+		airdropState     = "airdrop"
+		precompilesState = "precompiles"
+	)
 
 	var (
 		chainID    *big.Int
 		tokenName  string
 		vmVersion  string
 		allocation core.GenesisAlloc
-		direction  stateDirection
+		direction  statemachine.StateDirection
 		err        error
 	)
 
-	for stage != doneStage {
-		switch stage {
-		case startStage:
-			direction = forward
-		case descriptorStage:
+	subnetEvmState := statemachine.NewStateMachine(
+		[]string{descriptorsState, feeState, airdropState, precompilesState},
+	)
+	for subnetEvmState.Running() {
+		switch subnetEvmState.CurrentState() {
+		case descriptorsState:
 			chainID, tokenName, vmVersion, direction, err = getDescriptors(app, subnetEVMVersion)
-		case feeStage:
+		case feeState:
 			*conf, direction, err = getFeeConfig(*conf, app)
-		case airdropStage:
+		case airdropState:
 			allocation, direction, err = getAllocation(app, defaultEvmAirdropAmount, oneAvax, "Amount to airdrop (in AVAX units)")
-		case precompileStage:
+		case precompilesState:
 			*conf, direction, err = getPrecompiles(*conf, app)
 		default:
 			err = errors.New("invalid creation stage")
 		}
 		if err != nil {
 			return nil, nil, err
 		}
-		stage = nextStage(stage, direction)
+		subnetEvmState.NextState(direction)
```

Comment:
In fact we are not checking here for the return value and the possible return `""` (empty string), explicitly returning an error might make things clearer.

Reply:
I avoided this function to return anything

Reply:
But `NextState` _does_ return a string and it is not being checked here (it could be the empty string?)